orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/ManifestForceCacheRefreshTask.java

```diff
@@ -204,69 +199,63 @@ private boolean pendingRefreshProcessed(PendingRefresh pendingRefresh, Set<Strin
     }
   }
 
-  private Map<String, List<String>> manifestsNeedingRefresh(StageData stageData) {
-    Map<String, List<String>> deployedManifests = stageData.getManifestNamesByNamespace();
+  private List<ScopedManifest> manifestsNeedingRefresh(StageData stageData) {
+    List<ScopedManifest> deployedManifests = stageData.getDeployedManifests();
     Set<String> refreshedManifests = stageData.getRefreshedManifests();
     if (deployedManifests.isEmpty()) {
       log.warn("No manifests were deployed, nothing to refresh...");
     }
 
-    Map<String, List<String>> result = new HashMap<>();
-    for (Map.Entry<String, List<String>> entry : deployedManifests.entrySet()) {
-      String location = entry.getKey();
-      List<String> names = entry.getValue().stream()
-          .filter(n -> !refreshedManifests.contains(toManifestIdentifier(location, n)))
-          .collect(Collectors.toList());
-
-      if (!names.isEmpty()) {
-        result.put(location, names);
-      }
-    }
+    return deployedManifests.stream()
+      .filter(m -> !refreshedManifests.contains(toManifestIdentifier(m.location, m.name)))
+      .collect(Collectors.toList());
+  }
 
-    return result;
+  private List<ScopedManifest> getDeployedManifests(Stage stage) {
+    Map<String, List<String>> deployedManifests = manifestNamesByNamespace(stage);
+    return deployedManifests.entrySet().stream()
+      .flatMap(e -> e.getValue().stream().map(v -> new ScopedManifest(e.getKey(), v)))
+      .collect(Collectors.toList());
   }
 
   private boolean refreshManifests(String provider, String account, StageData stageData) {
-    Map<String, List<String>> manifests = manifestsNeedingRefresh(stageData);
-
-    final boolean[] allRefreshesSucceeded = {true};
-    for (Map.Entry<String, List<String>> entry : manifests.entrySet()) {
-      String location = entry.getKey();
-      entry.getValue().forEach(name -> {
-        String id = toManifestIdentifier(location, name);
-        Map<String, String> request = new ImmutableMap.Builder<String, String>()
-            .put("account", account)
-            .put("name", name)
-            .put("location", location)
-            .build();
-
-        try {
-          Response response = cacheService.forceCacheUpdate(provider, REFRESH_TYPE, request);
-          if (response.getStatus() == HTTP_OK) {
-            log.info("Refresh of {} in {} succeeded immediately", id, account);
-            stageData.getProcessedManifests().add(id);
-          } else {
-            allRefreshesSucceeded[0] = false;
-          }
-
-          stageData.getRefreshedManifests().add(id);
-        } catch (Exception e) {
-          log.warn("Failed to refresh {}: ", id, e);
-          allRefreshesSucceeded[0] = false;
-          stageData.errors.add(e.getMessage());
+    List<ScopedManifest> manifests = manifestsNeedingRefresh(stageData);
+
+    boolean allRefreshesSucceeded = true;
```

Comment:
I'm confused as to why `allRefreshesSucceeded` was previously `boolean[]`, if we were only every toggling the first element from true to false... :woman_shrugging:

Reply:
It was a workaround to the fact that you can only use a `final` or "effectively final" variable from within a lambda function.  So if we just used a `boolean` we'd need to set it to `final`, which of course means that we can't set it.  The workaround is to create a single-element array that is `final`...which makes the reference final, but still allows us to change the value it's pointing to.

I find this pattern pretty confusing (and also had no idea what it was doing the firs time I saw it)...removing it was one of the motivations behind my reworking some of this file.

Reply:
Whoah I never would have guessed that...thanks again for turning me reviewing your code into a java lesson for me!